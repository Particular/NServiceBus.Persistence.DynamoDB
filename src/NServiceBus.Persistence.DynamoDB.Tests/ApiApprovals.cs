namespace NServiceBus.Persistence.DynamoDB.Tests
{
    using NServiceBus;
    using NUnit.Framework;
    using Particular.Approvals;
    using PublicApiGenerator;

    [TestFixture]
    public class APIApprovals
    {
        [Test]
        public void Approve()
        {
            var publicApi = ApiGenerator.GeneratePublicApi(typeof(DynamoDBPersistence).Assembly, new ApiGeneratorOptions
            {
                ExcludeAttributes = new[] { "System.Runtime.Versioning.TargetFrameworkAttribute", "System.Reflection.AssemblyMetadataAttribute" }
            });
            Approver.Verify(publicApi);
        }
    }
}
