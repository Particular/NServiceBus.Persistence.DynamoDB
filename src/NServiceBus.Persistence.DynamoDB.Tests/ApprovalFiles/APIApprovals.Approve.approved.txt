[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"NServiceBus.Persistence.DynamoDB.AcceptanceTests, PublicKey=00240000048000009400000006020000002400005253413100040000010001007f16e21368ff041183fab592d9e8ed37e7be355e93323147a1d29983d6e591b04282e4da0c9e18bd901e112c0033925eb7d7872c2f1706655891c5c9d57297994f707d16ee9a8f40d978f064ee1ffc73c0db3f4712691b23bf596f75130f4ec978cf78757ec034625a5f27e6bb50c618931ea49f6f628fd74271c32959efb1c5")]
[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"NServiceBus.Persistence.DynamoDB.PersistenceTests, PublicKey=00240000048000009400000006020000002400005253413100040000010001007f16e21368ff041183fab592d9e8ed37e7be355e93323147a1d29983d6e591b04282e4da0c9e18bd901e112c0033925eb7d7872c2f1706655891c5c9d57297994f707d16ee9a8f40d978f064ee1ffc73c0db3f4712691b23bf596f75130f4ec978cf78757ec034625a5f27e6bb50c618931ea49f6f628fd74271c32959efb1c5")]
[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"NServiceBus.Persistence.DynamoDB.PessimisticLock.AcceptanceTests, PublicKey=00240000048000009400000006020000002400005253413100040000010001007f16e21368ff041183fab592d9e8ed37e7be355e93323147a1d29983d6e591b04282e4da0c9e18bd901e112c0033925eb7d7872c2f1706655891c5c9d57297994f707d16ee9a8f40d978f064ee1ffc73c0db3f4712691b23bf596f75130f4ec978cf78757ec034625a5f27e6bb50c618931ea49f6f628fd74271c32959efb1c5")]
[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"NServiceBus.Persistence.DynamoDB.Tests, PublicKey=00240000048000009400000006020000002400005253413100040000010001007f16e21368ff041183fab592d9e8ed37e7be355e93323147a1d29983d6e591b04282e4da0c9e18bd901e112c0033925eb7d7872c2f1706655891c5c9d57297994f707d16ee9a8f40d978f064ee1ffc73c0db3f4712691b23bf596f75130f4ec978cf78757ec034625a5f27e6bb50c618931ea49f6f628fd74271c32959efb1c5")]
[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"NServiceBus.Persistence.DynamoDB.TransactionalSession.AcceptanceTests, PublicKey=00240000048000009400000006020000002400005253413100040000010001007f16e21368ff041183fab592d9e8ed37e7be355e93323147a1d29983d6e591b04282e4da0c9e18bd901e112c0033925eb7d7872c2f1706655891c5c9d57297994f707d16ee9a8f40d978f064ee1ffc73c0db3f4712691b23bf596f75130f4ec978cf78757ec034625a5f27e6bb50c618931ea49f6f628fd74271c32959efb1c5")]
namespace NServiceBus
{
    public class DynamoPersistence : NServiceBus.Persistence.PersistenceDefinition { }
    public static class DynamoPersistenceConfig
    {
        public static void DisableTablesCreation(this NServiceBus.PersistenceExtensions<NServiceBus.DynamoPersistence> persistenceExtensions) { }
        public static NServiceBus.PersistenceExtensions<NServiceBus.DynamoPersistence> DynamoClient(this NServiceBus.PersistenceExtensions<NServiceBus.DynamoPersistence> persistenceExtensions, Amazon.DynamoDBv2.IAmazonDynamoDB dynamoClient) { }
        public static NServiceBus.Persistence.DynamoDB.SagaPersistenceConfiguration Sagas(this NServiceBus.PersistenceExtensions<NServiceBus.DynamoPersistence> persistenceExtensions) { }
        public static NServiceBus.PersistenceExtensions<NServiceBus.DynamoPersistence> UseSharedTable(this NServiceBus.PersistenceExtensions<NServiceBus.DynamoPersistence> persistenceExtensions, NServiceBus.Persistence.DynamoDB.TableConfiguration sharedTableConfiguration) { }
    }
    public interface IDynamoStorageSession
    {
        void Add(Amazon.DynamoDBv2.Model.TransactWriteItem writeItem);
        void AddRange(System.Collections.Generic.IEnumerable<Amazon.DynamoDBv2.Model.TransactWriteItem> writeItems);
    }
    public static class SynchronizedStorageSessionExtensions
    {
        public static NServiceBus.IDynamoStorageSession DynamoPersistenceSession(this NServiceBus.Persistence.ISynchronizedStorageSession session) { }
    }
}
namespace NServiceBus.Persistence.DynamoDB
{
    public static class DynamoOutboxConfigurationExtensions
    {
        public static NServiceBus.Outbox.OutboxSettings CreateTableWithInstaller(this NServiceBus.Outbox.OutboxSettings outboxSettings, bool createTable) { }
        public static NServiceBus.Outbox.OutboxSettings SetTimeToKeepDeduplicationData(this NServiceBus.Outbox.OutboxSettings outboxSettings, System.TimeSpan timeToLive) { }
        public static NServiceBus.Outbox.OutboxSettings UseTable(this NServiceBus.Outbox.OutboxSettings outboxSettings, System.Func<NServiceBus.Persistence.DynamoDB.TableConfiguration, NServiceBus.Persistence.DynamoDB.TableConfiguration> tableConfiguration) { }
    }
    public interface IDynamoClientProvider
    {
        Amazon.DynamoDBv2.IAmazonDynamoDB Client { get; }
    }
    public static class Mapper
    {
        public static System.Collections.Generic.Dictionary<string, Amazon.DynamoDBv2.Model.AttributeValue> ToMap(object value, System.Type type) { }
        public static System.Collections.Generic.Dictionary<string, Amazon.DynamoDBv2.Model.AttributeValue> ToMap<TValue>(TValue value)
            where TValue :  class { }
        public static object? ToObject(System.Collections.Generic.Dictionary<string, Amazon.DynamoDBv2.Model.AttributeValue> attributeValues, System.Type returnType) { }
        public static TValue? ToObject<TValue>(System.Collections.Generic.Dictionary<string, Amazon.DynamoDBv2.Model.AttributeValue> attributeValues) { }
    }
    public class OutboxPersistenceConfiguration
    {
        public OutboxPersistenceConfiguration() { }
        public NServiceBus.Persistence.DynamoDB.TableConfiguration Table { get; set; }
        public System.TimeSpan TimeToKeepDeduplicationData { get; set; }
    }
    public class SagaPersistenceConfiguration
    {
        public SagaPersistenceConfiguration() { }
        public System.TimeSpan LeaseAcquisitionTimeout { get; set; }
        public System.TimeSpan LeaseDuration { get; set; }
        public NServiceBus.Persistence.DynamoDB.TableConfiguration Table { get; set; }
        public bool UsePessimisticLocking { get; set; }
    }
    public class TableConfiguration : System.IEquatable<NServiceBus.Persistence.DynamoDB.TableConfiguration>
    {
        public TableConfiguration() { }
        public Amazon.DynamoDBv2.BillingMode BillingMode { get; set; }
        public string PartitionKeyName { get; set; }
        public Amazon.DynamoDBv2.Model.ProvisionedThroughput? ProvisionedThroughput { get; set; }
        public string SortKeyName { get; set; }
        public string TableName { get; set; }
        public string? TimeToLiveAttributeName { get; set; }
    }
}
namespace NServiceBus.Testing
{
    public class TestableDynamoSynchronizedStorageSession : NServiceBus.IDynamoStorageSession, NServiceBus.Persistence.ISynchronizedStorageSession
    {
        public TestableDynamoSynchronizedStorageSession() { }
        public System.Collections.Generic.IReadOnlyCollection<Amazon.DynamoDBv2.Model.TransactWriteItem> TransactWriteItems { get; }
    }
}